clusterDeployer.py

```diff
@@ -29,222 +29,654 @@
+        cmd = f"virsh net-update default add ip-dhcp-host \"{host_xml}\" --live --config"
+        self.hostconn.run_or_die(cmd)
+
+    def _ensure_started(self, api_network: str, bridge_xml: str) -> None:
+        cmd = "virsh net-destroy default"
+        self.hostconn.run(cmd)  # ignore return code - it might fail if net was not started
+
+        cmd = "virsh net-undefine default"
+        ret = self.hostconn.run(cmd)
+        if ret.returncode != 0 and "Network not found" not in ret.err:
+            logger.error_and_exit(str(ret))
+
+        # Fix cases where virsh net-start fails with error "... interface virbr0: File exists"
+        cmd = "ip link delete virbr0"
+        self.hostconn.run(cmd)  # ignore return code - it might fail if virbr did not exist
+
+        cmd = f"virsh net-define {bridge_xml}"
+        self.hostconn.run_or_die(cmd)
+
+        # set interface down before starting bridge as otherwise bridge start might fail if interface
+        # already got an IP address in same network as bridge
+        self.hostconn.run(f"ip link set {api_network} down")
+
+        cmd = "virsh net-start default"
+        self.hostconn.run_or_die(cmd)
+
+        self.hostconn.run(f"ip link set {api_network} up")
+
+    def limit_dhcp_range(self, old_range: str, new_range: str) -> None:
+        # restrict dynamic dhcp range: we use static dhcp ip addresses; however, those addresses might have been used
+        # through the dynamic dhcp by any systems such as systems ready to be installed.
+        cmd = "virsh net-dumpxml default"
+        ret = self.hostconn.run(cmd)
+        if f"range start='{old_range}'" in ret.out:
+            host_xml = f"<range start='{old_range}' end='192.168.122.254'/>"
+            cmd = f"virsh net-update default delete ip-dhcp-range \"{host_xml}\" --live --config"
+            r = self.hostconn.run(cmd)
+            logger.debug(r.err if r.err else r.out)
+
+            host_xml = f"<range start='{new_range}' end='192.168.122.254'/>"
+            cmd = f"virsh net-update default add ip-dhcp-range \"{host_xml}\" --live --config"
+            r = self.hostconn.run(cmd)
+            logger.debug(r.err if r.err else r.out)
+
+    def _network_xml(self, ip: str, dhcp_range: Optional[Tuple[str, str]] = None) -> str:
+        if dhcp_range is None:
+            dhcp_part = ""
+        else:
+            dhcp_part = f"""<dhcp>
+                <range start='{dhcp_range[0]}' end='{dhcp_range[1]}'/>
+                </dhcp>"""
+
+        return f"""
+                <network>
+                <name>default</name>
+                <forward mode='nat'/>
+                <bridge name='virbr0' stp='off' delay='0'/>
+                <ip address='{ip}' netmask='255.255.0.0'>
+                    {dhcp_part}
+                </ip>
+                </network>"""
+
+    def configure(self, api_network: str) -> None:
+        hostname = self.hostconn.hostname()
+        cmd = "systemctl enable libvirtd"
```

Comment:
`systemctl enable libvirtd --now` and remove the start below

Reply:
Ack.